Lib/idlelib/sidebar.py

```diff
@@ -40,6 +54,15 @@ def get_widget_padding(widget):
     return padx, pady
 
 
+@contextlib.contextmanager
+def temp_enable_text_widget(text):
+    text.configure(state=tk.NORMAL)
+    try:
+        yield
+    finally:
+        text.configure(state=tk.DISABLED)
+
+
 class BaseSideBar:
     """
     The base class for extensions which require a sidebar.
```

Comment:
'extensions is not the right word anymore.

I am stopping here for now.

Reply:
@terryjreedy:
> 'extensions is not the right word anymore.

Would the correct term here be "derived class"? That seems to be the term used in https://docs.python.org/3.9/tutorial/classes.html, but there may a better word to use that would be more specific to this context. If "extensions" is no longer the correct term to use, the documentation for IDLE should probably be updated accordingly https://docs.python.org/3.9/library/idle.html?highlight=extensions. 

Also, minor grammar nit: "that" is more correct than "which" in this context. 

```suggestion
     The base class for derived classes that require a sidebar.
```

Reply:
See `idlelib/config_extensions.def` for the meaning of 'extensions'.  Many IDLE features were once implemented as extensions but no longer.